flax/nnx/transforms/iteration.py

```diff
@@ -767,6 +769,12 @@ def _scan_split_in(
     if not vectorized_states:
       vectorized_states.append(State({}))
     carry_deque.append(carry_states)
+    carry_graphdefs_deque.append(graphdef)
+    if len(carry_states) > 0:
+      # We should not store carry graphdef in the args as scan call
+      # can change the graphdef of the output carry.
+      # Thus we keep it in carry_graphdefs_deque
+      graphdef = None  # type: ignore[assignment]
```

Comment:
can we can remove this conditional and just pass `None` to `NodeStates.from_split`?